frontend/src/citizen-frontend/pedagogical-documents/PedagogicalDocuments.tsx

```diff
@@ -26,132 +26,132 @@ import {
-  ])
+  const toggleExpanded = useCallback(() => {
+    setExpanded((prev) => !prev)
+  }, [])
 
-  const { setErrorMessage } = useContext(OverlayContext)
+  // A description with more than 50 characters per line will be collapsed
+  const shouldShowExpandButton =
+    item?.description && item.description.length > 50 * clampLines
 
-  const onAttachmentUnavailable = useCallback(
-    () =>
-      setErrorMessage({
-        title: t.fileDownload.modalHeader,
-        text: t.fileDownload.modalMessage,
-        type: 'error'
-      }),
-    [t, setErrorMessage]
+  return (
+    <FixedSpaceRow spacing="xs" alignItems="end">
+      <ExpandableText
+        expanded={expanded}
+        clampLines={clampLines}
+        data-qa={dataQa}
+      >
+        {item.description}
+      </ExpandableText>
+      {shouldShowExpandButton && (
+        <IconButton
+          onClick={toggleExpanded}
+          data-qa={`${dataQa}-button`}
+          icon={expanded ? faChevronUp : faChevronDown}
+          altText={t.pedagogicalDocuments.toggleExpandText}
+        />
+      )}
+    </FixedSpaceRow>
   )
+})
 
-  const onRead = (doc: PedagogicalDocumentCitizen) => {
-    void markPedagogicalDocumentRead(doc.id).then(loadData)
-  }
-
-  const Attachment = ({
-    item,
-    dataQa
+const PedagogicalDocumentsDisplay = React.memo(
+  function PedagogicalDocumentsDisplay({
+    items,
+    onRead,
+    onAttachmentUnavailable
   }: {
-    item: PedagogicalDocumentCitizen
-    dataQa: string
-  }) => {
-    return (
-      <>
-        {item && item.attachment && (
-          <FileDownloadButton
-            key={item.attachment.id}
-            file={item.attachment}
-            fileFetchFn={getAttachmentBlob}
-            afterFetch={() => onRead(item)}
-            onFileUnavailable={onAttachmentUnavailable}
-            icon
-            data-qa={dataQa}
-            openInBrowser={true}
-          />
-        )}
-      </>
-    )
-  }
+    items: PedagogicalDocumentCitizen[]
+    onRead: (doc: PedagogicalDocumentCitizen) => void
+    onAttachmentUnavailable: () => void
+  }) {
+    const t = useTranslation()
 
-  const AttachmentDownloadButton = ({
-    item,
-    dataQa
-  }: {
-    item: PedagogicalDocumentCitizen
-    dataQa: string
-  }) => {
-    return (
-      <>
-        {item.attachment && (
-          <FileDownloadButton
-            key={item.attachment.id}
-            file={item.attachment}
-            fileFetchFn={getAttachmentBlob}
-            afterFetch={() => onRead(item)}
-            onFileUnavailable={onAttachmentUnavailable}
-            icon={faArrowDown}
-            data-qa={dataQa}
-            text={t.fileDownload.download}
-          />
-        )}
-      </>
+    const countChildren = useCallback(
+      () =>
+        items.reduce(
+          (childIds, doc) =>
+            childIds.includes(doc.childId)
+              ? childIds
+              : [...childIds, doc.childId],
+          [] as string[]
+        ).length,
+      [items]
```

Comment:
We have `lodash` if you prefer something like `_.uniq(items.map(doc => doc.childId)).length`

Reply:
This could actually be changed to `const moreThanOneChild = useMemo(...)` because it's called right away